app/services/gogox.rb

```diff
@@ -0,0 +1,129 @@
+class Gogox
```

Comment:
Just thing which can be done, no compulsoion

```ruby
class Gogox < BaseTransportation
   # it inherits a base class
end
```

```ruby
class BaseTransportation
   # Common code goes in here which would be applicable to almost all the types, for ex:

  def parse_pickup_time(time = nil)
    return time if time.present?

    # next available date within next 5 days
    next_available_date = DateSet.new(5, 1).available_dates.first
    (next_available_date.beginning_of_day + 12.hours)
  end

  def parse_time
    # logic follows
  end

   def book
     # raise error if child class doesnt provide implementation for it
     raise NotImplementedError
   end
end
```

Reply:
@bharat619 you are suggesting to use template design pattern here, which seems valid in this case.
But I have not added one more abstraction layer here since we are not yet sure about how the other providers' API is, do they work in the same manner or there are different.

Once we have at least one more provider API available, we can introduce above suggestion.